tgui/packages/tgui/interfaces/ModularShieldGen.tsx

```diff
@@ -0,0 +1,142 @@
+import { useBackend } from '../backend';
+import { Window } from '../layouts';
+import { Stack, Section, ProgressBar, Button, NumberInput, LabeledList } from '../components';
+import { BooleanLike } from 'common/react';
+
+type ModularShieldGenData = {
+  max_strength: number;
+  current_strength: number;
+  max_regeneration: number;
+  current_regeneration: number;
+  max_radius: number;
+  current_radius: number;
+  active: BooleanLike;
+  recovering: BooleanLike;
+  exterior_only: BooleanLike;
+  initiating_field: BooleanLike;
+};
+
+export const ModularShieldGen = (props, context) => {
+  const { topLevel } = props;
+  const { act, data } = useBackend<ModularShieldGenData>(context);
+  const {
+    max_strength,
+    max_regeneration,
+    current_regeneration,
+    max_radius,
+    current_radius,
+    current_strength,
+    active,
+    exterior_only,
+    recovering,
+    initiating_field,
+  } = data;
+
+  return (
+    <Window title="Modular Shield Generator" width={600} height={400}>
+      <Window.Content scrollable>
+        <Section title={'Shield Strength'} color={recovering ? 'red' : 'white'}>
+          <ProgressBar
+            title="Shield Strength"
+            value={current_strength}
+            minValue={0}
+            maxValue={max_strength}
+            height={'30px'}
+            ranges={{
+              'good': [max_strength * 0.75, max_strength],
+              'average': [max_strength * 0.25, max_strength * 0.75],
+              'bad': [0, max_strength * 0.25],
+            }}>
+            {current_strength}/{max_strength}
+          </ProgressBar>
+        </Section>
+        <Section title={'Regeneration and Radius'}>
+          <ProgressBar
+            height={'20px'}
+            title="Regeneration rate"
+            value={current_regeneration}
+            minValue={0}
+            maxValue={max_regeneration}
+            ranges={{
+              'good': [max_regeneration * 0.75, max_regeneration],
+              'average': [max_regeneration * 0.25, max_regeneration * 0.75],
+              'bad': [0, max_regeneration * 0.25],
+            }}>
+            Regeneration {current_regeneration}/{max_regeneration}
+          </ProgressBar>
+          <Section>
+            <ProgressBar
+              height={'20px'}
+              title="Shield radius"
+              value={current_radius}
+              minValue={0}
+              maxValue={max_radius}
+              ranges={{
+                'good': [max_radius * 0.75, max_radius],
+                'average': [max_radius * 0.25, max_radius * 0.75],
+                'bad': [0, max_radius * 0.25],
+              }}>
+              Radius {current_radius}/{max_radius}
+            </ProgressBar>
+          </Section>
+        </Section>
+        <Section title={'Settings'}>
+          <Stack>
+            <Section>
+              <LabeledList>
+                <LabeledList.Item label="Set Radius">
+                  <Section right={0.75}>
```

Comment:
```suggestion
```

Reply:
how will i adjust the placement of the num input?